scipy/signal/tests/test_peak_finding.py

```diff
@@ -577,12 +602,12 @@ def test_width_condition(self):
         """
         x = np.array([1, 0, 1, 2, 1, 0, -1, 4, 0])
         peaks, props = find_peaks(x, width=(None, 2), rel_height=0.75)
-        assert_(peaks.size == 1)
-        assert_(peaks == 7)
-        assert_(props['widths'] == 1.35)
-        assert_(props['width_heights'] == 1.)
-        assert_(props['left_ips'] == 6.4)
-        assert_(props['right_ips'] == 7.75)
+        assert_equal(peaks.size, 1)
```

Comment:
Can you clarify why we want to migrate `assert_` to `assert_equal`; there has already been some discussion of `assert_` vs. plain `assert` in the pytest context.

Reply:
The motivation would be similar in this case, I think. `assert_` does not give as informative a message as either `assert ` or `assert_equal`. And the latter can be used while we settle #8583

Reply:
Yeah, that's exactly it. 

To elaborate why I'm using `assert_almost_equal` a few lines below: because of numeric errors when subtracting floats the statements need to be more tolerant. Previously the difference was calculated inside a window with adjustment of the index values (`some_index - window_start`) which isn't necessary with the proposed changes. As a consequence, a width that was previously calculated as 
```python
>>> 1.75 - 0.4
1.35
```
is now calculated with
```python
>>> 7.75 - 6.4
1.3499999999999996
```

Reply:
Will do.